cluster/examples/kubernetes/ceph/common.yaml

```diff
@@ -1476,6 +1476,22 @@ rules:
   verbs:
   - get
 ---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: rook-ceph-multus
+  namespace: default
+  labels:
+    operator: rook
+    storage-backend: ceph
+rules:
+- apiGroups:
+  - k8s.cni.cncf.io
+  resources:
+  - network-attachment-definitions
+  verbs:
+  - get
```

Comment:
Looks like we're calling `Get()` on this in two places, one for validation, and the other to get the actual configuration. But that code already existed before this PR and we didn't have the RBAC. Was the RBAC just overlooked before? 

Even better, if there is any way to avoid calling `Get()` so we can avoid this RBAC change. For example, if we're only using it for validation, I would suggest we don't need it.

Reply:
Yes, we just need it to check if the NAD is present or not.
And the previous RBAC didn't work when the NAD was in diffrent namespace.

Reply:
I will look into fine tuning the RBAC.

Reply:
If RBAC is only being used for validation, my view is still that we should skip the validation and just try to use it. If the NAD does not exist, the pod will just stay pending until it does exist. This is fairly standard K8s behavior to attempt and action and retry until it succeeds. If they realize there is a problem and used the wrong NAD, they can update the CR and then the pod will be updated.

Reply:
Okay

Reply:
@leseb I will remove the validation

Reply:
Has this been addressed?

Reply:
@travisn I'm in favor of keeping the early validation. The fail-early approach actually encapsulates the "things eventually succeed k8s philosophy". If we fail early, the user gets a clear message on what's going on and our reconcile kicks in again with no pending resources. If we do nothing then resources will be pending, it's fine it's just that the operator logs won't give any useful info. From a deployment standpoint I feel like the fail early is more appropriate.

Reply:
@rohan47 What exactly is the behavior if the NAD does not exist? What failures will be observed?

Reply:
The validation will return an error that the NAD is not present and CephCluster creation will fail @travisn

Reply:
To be clear:
1. If we add validation in Rook, we will see errors in the operator log and the cluster creation will fail. However, this is not done as part of the admission controller, so the admin will have to find errors in the operator logs.
2. If we don't add validation in Rook, what is the behavior? What does the failure look like?

Reply:
> To be clear:
> 
>     1. If we add validation in Rook, we will see errors in the operator log and the cluster creation will fail. This is not done as part of the admission controller, so the admin will have to find errors in the operator logs.
> 
>     2. If we don't add validation in Rook, what is the behavior? What does the failure look like?

If we don't add validation in Rook, the cluster creation process will continue and will get stuck at Mon pods in pending state as they will be waiting for the NAD to be present.

In both the cases the cluster creation will resume after NADs are created.

Just confirmed that the CSI pods start getting created and get stuck in pending state if NADs are not present and they are out of validation scope as of now.

Reply:
Ok so the mon pod(s) will remain in pending state, which should actually be more easily discoverable. It's easier to notice a pending pod than to find an error in the operator log.

Reply:
Yes, it makes sense.
@leseb do we agree to remove the validations?

Reply:
We need to get the NAD to get the subnet to pass it to Ceph config.
Sorry for overlooking this earlier and the confusion around this @travisn @leseb 
So I am removing the validation but we still need the RBAC.

Reply:
Ok, so we anyway need to look up the NAD so we can get the needed subnet config. Then sounds good to add the RBAC and fail the reconcile as originally implemented.

Reply:
Thinking about upgrades, if the RBAC hasn't been created, nothing will fail in the reconcile unless multus settings are enabled, right? Just wanted to verify that non-multus clusters won't be affected.

Reply:
Yes, non-multus cluster won't be affected.
While creating a multus cluster, if the RBAC is not created then while getting the NAD it will fail with the following error
```
failed to fetch network attachment definition for selector "public":
network-attachment-definitions.k8s.cni.cncf.io "public-nad" is forbidden: 
User "system:serviceaccount:rook-ceph:rook-ceph-system" cannot get resource 
"network-attachment-definitions" in API group "k8s.cni.cncf.io" in the namespace "default"
```

Reply:
Removed the unnecessary RBAC and tested the changes.